dipy/workflows/base.py

```diff
@@ -101,19 +101,33 @@ def add_workflow(self, workflow):
             ref_text = [text if text else "\n" for text in npds['References']]
             ref_idx = self.epilog.find('References: \n') + len('References: \n')
             self.epilog = "{0}{1}\n{2}".format(self.epilog[:ref_idx],
-                                               ''.join([text for text in ref_text]),
+                                        ''.join([text for text in ref_text]),
                                                self.epilog[ref_idx:])
 
         self.outputs = [param for param in npds['Parameters'] if
                         'out_' in param[0]]
 
         args, defaults = get_args_default(workflow.run)
 
+        output_args = \
+            self.add_argument_group('output arguments(optional)')
+
         len_args = len(args)
         len_defaults = len(defaults)
 
-        output_args = \
-            self.add_argument_group('output arguments(optional)')
+        #  This check simply shows a helpful message to the user if there
+        #  is a mismatch in the number of arguments in the run method and
+        #  the doc string. Doc string refers to the parameter help written
+        # in the workflow python script.
+
+        if len_args != len(self.doc):
+            raise ValueError(
+                            self.prog +
+                            ": Number of parameters in the "
+                            "doc string and run method does not match. "
+                            "Please ensure that the number of parameters "
+                            "in the run method is same as the doc string.")
```

Comment:
Comments above and this text agree, which is good, but what's the point of the comment then? We usually add comments to explain something more clearly. IMO, it's not required here.

Also, your indentation is weird here.

Reply:
Hey,

Thanks for the comments. I will discuss this with the team and remove the comments. So the reason for putting the comment is to ensure some space in case the message needs to be changed in future (a new developer to DIPY, like me) will find the comment useful to understand the piece of code.

About the indentation, I made it strictly compliant to PEP8 since the dry run of PEP8 was complaining about the under or over indentation. So I made it strictly according to the PEP8 standards. 

I agree with you, the indentation is not good enough but since the change was in the core file of DIPY workflows so I chose to follow the PEP8 coding style to avoid PEP8 messages in the PR checks.

Thanks :)

Reply:
Hey,

I updated the indentation.

Thanks.